src/Voyager.php

```diff
@@ -19,16 +21,64 @@ class Voyager
 
     protected $alertsCollected = false;
 
+    protected $formFields = [];
+    protected $afterFormFields = [];
+
     public function __construct()
     {
         $this->filesystem = app(Filesystem::class);
 
         $this->findVersion();
     }
 
+    public function formField($row, $dateType, $dataTypeContent)
+    {
+        $formField = $this->formFields[$row->type];
+
+        return $formField->handle($row, $dateType, $dataTypeContent);
+    }
+
+    public function afterFormFields($row, $dataType, $dataTypeContent)
+    {
+        $options = json_decode($row->details);
+
+        return collect($this->afterFormFields)->filter(function ($after) use ($row, $dataType, $dataTypeContent, $options) {
+            return $after->visible($row, $dataType, $dataTypeContent, $options);
+        });
+    }
+
+    public function addFormField($handler)
+    {
+        if (!$handler instanceof HandlerInterface) {
+            $handler = app($handler);
+        }
+
+        $this->formFields[$handler->getCodename()] = $handler;
```

Comment:
`return $this` maybe? So you can `addFormField()->addAfterFormField()->addAfterFormField()`